base-client/src/main/java/no/ebax/sdk/baseclient/ApiClient.java

```diff
@@ -0,0 +1,10 @@
+package no.ebax.sdk.baseclient;
+
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Flow;
+
+public interface ApiClient {
+    <T> Flow.Publisher<T> post(String uri, Flow.Publisher<String> bodyPublisher, Map<String, List<String>> headers, Class<T> tClass);
```

Comment:
And would it make sense to not delegate the responsibility for unmarshalling response bodies? It already expects marshalled request bodies, so it would be symmetric, and the responsibility for the provider would be smaller.